tests/upgrades/test_contentview.py

```diff
@@ -6,253 +6,177 @@
-            self.yum_repo1['name'],
-            'Repo was not associated to CV',
-        )
+        test_name = request.node.name
+        puppet_module = {'name': 'versioned', 'version': '3.3.3'}
+        org = entities.Organization(name=f'{request.node.name}_org').create()
+        product = entities.Product(organization=org, name=f'{request.node.name}_prod').create()
+        yum_repository = entities.Repository(
+            product=product, name=f'{test_name}_yum_repo', url=FAKE_1_YUM_REPO
+        ).create()
+        entities.Repository.sync(yum_repository)
+        puppet_repository = entities.Repository(
+            product=product,
+            name=f'{request.node.name}_puppet_repo',
+            content_type="puppet",
+            url=CUSTOM_PUPPET_REPO,
+        ).create()
+        entities.Repository.sync(puppet_repository)
+        puppet_module_list = PuppetModule.list(
+            {'search': 'name={name} and version={version}'.format(**puppet_module)}
+        )[0]
+
+        file_repository = entities.Repository(
+            product=product, name=f'{test_name}_file_repo', content_type="file"
+        ).create()
+
+        remote_file_path = f"/tmp/{RPM_TO_UPLOAD}"
+
+        ssh.upload_file(local_file=get_data_file(RPM_TO_UPLOAD), remote_file=remote_file_path)
+        with open(f'{get_data_file(RPM_TO_UPLOAD)}', "rb") as content:
+            file_repository.upload_content(files={'content': content})
+        assert RPM_TO_UPLOAD in file_repository.files()["results"][0]['name']
+        cv = entities.ContentView(name=f"{test_name}_cv", organization=org).create()
+        cv.repository = [yum_repository, file_repository]
+        cv.update(['repository'])
         ContentView.puppet_module_add(
             {
-                'content-view-id': content_view['id'],
-                'name': self.puppet_module['name'],
-                'author': self.puppet_module['author'],
+                'content-view-id': cv.id,
+                'name': puppet_module_list['name'],
+                'author': puppet_module_list['author'],
             }
         )
-        content_view = ContentView.info({'id': content_view['id']})
-        self.assertGreater(len(content_view['puppet-modules']), 0)
-        ContentView.publish({'id': content_view['id']})
-        cv = ContentView.info({'id': content_view['id']})
-        self.assertEqual(len(cv['versions']), 1)
+        cv.publish()
+        assert len(cv.puppet_module) == 0
+        assert len(cv.read_json()['versions']) == 1
 
     @post_upgrade(depend_on=test_cv_preupgrade_scenario)
-    def test_cv_postupgrade_scenario(self):
-        """This is post-upgrade scenario test to verify if we can update
-        content-view created in pre-upgrade scenario with various repositories.
+    def test_cv_postupgrade_scenario(self, request, dependent_scenario_name):
+        """After upgrade, Content-view update should work on existing content-view(created before
```

Comment:
```suggestion
        """After upgrade, the existing content-view(created before upgrade) should be updated.
```
Since, `should work` is very general testing term which doesn't feels good in docs like test description here.

Reply:
fixed.